src/python_testing/TC_FAN_3_5.py

```diff
@@ -0,0 +1,209 @@
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, expected_status, "Unexpected error returned")
+            pass
+
+    @async_test_body
+    async def test_TC_FAN_3_5(self):
+        if not self.check_pics("FAN.S.F04"):
+            logger.info("Test skipped because PICS FAN.S.F04 is not set")
+            return
+
+        endpoint = self.user_params.get("endpoint", 1)
+
+        # Part 1
+
+        self.print_step(1, "Commissioning, already done")
+
+        # Part 2
+
+        self.print_step("2a", "writes to the DUT the PercentSetting attribute with 50")
+        await self.write_percent_setting(endpoint=endpoint, percent_setting=50)
+
+        time.sleep(1)
+
+        self.print_step("2b", "Read from the DUT the PercentCurrent attribute and store")
+        percent_current = await self.read_percent_current(endpoint=endpoint)
+
+        self.print_step("2c", "TH sends Step command to DUT with Direction set to Increase")
+        await self.send_step_command(endpoint=endpoint, direction=Clusters.Objects.FanControl.Enums.StepDirectionEnum.kIncrease)
+
+        time.sleep(1)
+
+        self.print_step("2d", "Read from the DUT the PercentCurrent attribute and check its higher than the stored value")
+        percent_current_after = await self.read_percent_current(endpoint=endpoint)
+        asserts.assert_greater(percent_current_after, percent_current, "PercentCurrent did not increase")
+
+        # Part 3
+
+        self.print_step("3a", "writes to the DUT the PercentSetting attribute with 50")
+        await self.write_percent_setting(endpoint=endpoint, percent_setting=50)
+
+        time.sleep(1)
+
+        self.print_step("3b", "Read from the DUT the PercentCurrent attribute and store")
+        percent_current = await self.read_percent_current(endpoint=endpoint)
+
+        self.print_step("3c", "TH sends Step command to DUT with Direction set to Decrease")
+        await self.send_step_command(endpoint=endpoint, direction=Clusters.Objects.FanControl.Enums.StepDirectionEnum.kDecrease)
+
+        self.print_step("3d", "Read from the DUT the PercentCurrent attribute and check its lower than the stored value")
+        percent_current_after = await self.read_percent_current(endpoint=endpoint)
+        asserts.assert_less(percent_current_after, percent_current, "PercentCurrent did not decrease")
+
+        # Part 4
+        self.print_step("4a", "TH reads from the DUT the SpeedMax attribute")
+        speed_max = await self.read_speed_max(endpoint=endpoint)
+
+        self.print_step("4b", "TH writes to the DUT the SpeedSetting attribute with the value of SpeedMax")
+        await self.write_speed_setting(endpoint=endpoint, speed_setting=speed_max)
+
+        time.sleep(1)
+
+        self.print_step("4c", "TH sends Step command to DUT with Direction set to Increase and Wrap set to False")
```

Comment:
General Q: Does this command return an error here if it can't do what you're asking? If so, the test plan should add that. If not...should it? Does a client need to know that this won't do anything?

Reply:
It's loosely worded in the spec because the entire command is application specific but when it came up in the past, we generally agreed it was a successful command that had no effect. 

I will raise an issue on the spec repo for it for next release.